lib/pluginmanager/install.rb

```diff
@@ -59,9 +59,30 @@ def execute
 
     install_gems_list!(gems)
     remove_unused_locally_installed_gems!
+    remove_unused_integration_overlaps!
   end
 
   private
+
+  def remove_unused_integration_overlaps!
+    installed_plugin_specs = plugins_arg.map do |plugin_arg|
+      if LogStash::PluginManager.plugin_file?(plugin_arg)
+        LogStash::PluginManager.plugin_file_spec(plugin_arg)
+      else
+        LogStash::PluginManager.find_plugins_gem_specs(plugin_arg)
+      end
+    end.flatten.select do |spec|
```

Comment:
instead of chaining `Enumerable#map` and `Enumerable#flatten`, we can use `Enumerable#flat_map`